### PR TITLE
auth service on init

### PR DESCRIPTION
**Pull request by recca0120**

### CODE REVIEW

File: src/app/core/authentication/auth.service.ts [hunk 1]
```diff
@@ -19,8 +19,8 @@ export class AuthService {
 
   constructor(private loginService: LoginService, private tokenService: TokenService) {}
 
-  onChange() {
-    return this.change$;
+  onInit(): void {
```
Comment (nzbin) [inline src/app/core/authentication/auth.service.ts:22]: Is `init` better than `onInit`?

File: src/app/core/bootstrap/startup.service.ts [hunk 2]
```diff
@@ -23,10 +22,10 @@ export class StartupService {
   load() {
     return new Promise((resolve, reject) => {
       this.authService
-        .onChange()
+        .user()
```
Comment (nzbin) [inline src/app/core/bootstrap/startup.service.ts:25]: Will it be better to set a generics type?

File: src/app/core/authentication/auth.service.ts (hunk 1 above)
Comment (recca0120) [reply src/app/core/authentication/auth.service.ts:22]: Yes

File: src/app/core/bootstrap/startup.service.ts (hunk 2 above)
Comment (recca0120) [reply src/app/core/bootstrap/startup.service.ts:25]: Any suggestions?
I will change it to <User|null>

Comment (nzbin) [reply src/app/core/bootstrap/startup.service.ts:25]: `this.authService.user<User>()`

If we can set user method a generics type, the developer don't need to modify the auth service.

Comment (recca0120) [reply src/app/core/bootstrap/startup.service.ts:25]: Could you give me a example ?

Comment (nzbin) [reply src/app/core/bootstrap/startup.service.ts:25]:
```
user<T=User>(): Observable<T> {
  return this.user$.pipe(share());
}

this.authService.user<CustomUser>()
```

In a real app, the user type maybe different.

Comment (recca0120) [reply src/app/core/bootstrap/startup.service.ts:25]: user() can return any?

Comment (recca0120) [reply src/app/core/bootstrap/startup.service.ts:25]: like this https://stackoverflow.com/questions/49622045/in-typescript-what-does-t-mean

**Comment by nzbin** [comment]
Why I think the commit (8e3bc7f1e304fb7358acc66f811206cf86c53bda) is more complicated, please rebase it, I will have a try. 🤣

**Pull request merged**